@here/harp-omv-datasource/lib/OmvData.ts

```diff
@@ -576,41 +580,121 @@ export class OmvProtobufDataAdapter implements OmvDataAdapter, OmvVisitor {
+                                command.longitude
+                            );
+                            currentRing = { coordinates: [geoPoint], outlines: [command.outline!] };
+                        } else if (isLineToCommand(command)) {
+                            const geoPoint = new GeoCoordinates(
+                                command.latitude,
+                                command.longitude
+                            );
+                            currentRing.coordinates.push(geoPoint);
+                            currentRing.outlines!.push(command.outline!);
+                        } else if (isClosePathCommand(command)) {
+                            currentPolygon.rings.push(currentRing);
+                        }
                     }
                 }
-            }
-        );
+            );
 
-        if (currentPolygon.rings.length > 0) {
-            geometry.push(currentPolygon);
-        }
+            if (currentPolygon.rings.length > 0) {
+                geometry.push(currentPolygon);
+            }
 
-        if (geometry.length === 0) {
-            return;
+            if (geometry.length > 0) {
+                this.m_processor.processPolygonFeature(layerName, geometry, env, storageLevel);
+            }
         }
 
-        const env = createFeatureEnv(this.m_layer, feature, "polygon", storageLevel, this.m_logger);
+        if (includesLineTechnique) {
+            const geometry: ILineGeometry[] = [];
+            // const currentPolygon: IPolygonGeometry = { rings: [] };
+            let currentRing: IRing;
+            this.m_geometryCommands.accept(
+                feature.geometry,
+                this.m_tileKey,
+                this.m_geoBox,
+                this.m_layer.extent,
+                {
+                    type: "Polygon",
+                    visitCommand: command => {
+                        if (isMoveToCommand(command)) {
+                            const geoPoint = new GeoCoordinates(
+                                command.latitude,
+                                command.longitude
+                            );
+                            currentRing = { coordinates: [geoPoint], outlines: [command.outline!] };
+                        } else if (isLineToCommand(command)) {
+                            const geoPoint = new GeoCoordinates(
+                                command.latitude,
+                                command.longitude
+                            );
+                            currentRing.coordinates.push(geoPoint);
+                            currentRing.outlines!.push(command.outline!);
+                        } else if (isClosePathCommand(command)) {
+                            if (currentRing.outlines === undefined) {
+                                geometry.push({ coordinates: currentRing.coordinates });
+                            } else {
+                                let line: ILineGeometry = { coordinates: [] };
```

Comment:
Why you postprocess `rings` when you could create line geometries directly in commands handler ?